skrub/_agg_joiner.py

```diff
@@ -229,17 +336,17 @@ def transform(self, X):
         X_transformed : DataFrameLike
             The augmented input.
         """
-
         check_is_fitted(self, "aux_table_")
-        skrub_px, _ = get_df_namespace(*[aux_table for aux_table, _ in self.aux_table_])
-
-        for aux_table, aux_key in self.aux_table_:
-            X = skrub_px.join(
-                left=X,
-                right=aux_table,
-                left_on=self.main_key_,
-                right_on=aux_key,
-            )
+        X, _ = self._check_dataframes(X, self.aux_table_)
+        _join_utils.check_missing_columns(X, self._main_key, "'X' (the main table)")
+
+        skrub_px, _ = get_df_namespace(self.aux_table_)
+        X = skrub_px.join(
+            left=X,
+            right=self.aux_table_,
+            left_on=self._main_key,
+            right_on=self._aux_key,
+        )
```

Comment:
I can't comment on it below but let's add the function `_screen` when we actually have something to put in there, we can remove it for now